plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -136,7 +136,8 @@ public class DefaultJdbcMetadata
     private final boolean precalculateStatisticsForPushdown;
     private final Set<JdbcQueryEventListener> jdbcQueryEventListeners;
 
-    private final AtomicReference<Runnable> rollbackAction = new AtomicReference<>();
+    private final AtomicReference<Runnable> insertRollbackAction = new AtomicReference<>();
+    private final AtomicReference<MergeRollbackAction> mergeRollbackAction = new AtomicReference<>();
```

Comment:
I think it will be more readable if you just replace
```
private final AtomicReference<Runnable> insertRollbackAction = new AtomicReference<>();
```

with 

```
private List<Runnable> rollbackActions;
```

Is there huge benefit of having this as `AtomicReference`? I am not sold on that.

Reply:
I see that the `DefaultJdbcMetadataFactory` will always create the metadata, so it seems unlikely that it could run into a race condition. Updated.